addons/mail/static/src/models/sender/sender.js

```diff
@@ -0,0 +1,141 @@
+/** @odoo-module **/
+
+import { registerModel } from '@mail/model/model_core';
+import { attr, many, one } from '@mail/model/model_field';
+import { link, clear, insertAndReplace } from '@mail/model/model_field_command';
+
+registerModel({
+    name: 'Sender',
+    identifyingFields: ['thread'],
+    recordMethods: {
+        /**
+         * Insert a new message composition inside the message queue and start
+         * to send message.
+         *
+         * @param {MessageComposition} composition
+         */
+        async queueMessage(compositionData) {
+            const composition = this.messaging.models['MessageComposition'].insert(compositionData);
+            this.update({ messagesToSend: link(composition) });
+            if (composition.composerView.threadView) {
+                composition.composerView.threadView.addComponentHint(
+                    'message-received',
+                    { message: composition.message }
+                );
+            }
+            // Prevent multiple queue to run at the same time.
+            if (this.isSending) {
+                return;
+            }
+            this._processQueue();
+        },
+        async _processQueue() {
+            this.update({ isSending: true });
+            const composition = this.messagesToSend[0];
+            await this.postMessage(composition);
+            if (this.messagesToSend.length > 0) {
+                return this._processQueue();
+            }
+            return this.update({ isSending: false });
+        },
+        /**
+         * Post a message in provided composer's thread based on current
+         * composer fields values.
+         */
+        async postMessage(composition) {
```

Comment:
I don't like this method. Too long, too many variables, etc.
This model has a poor balance of fields/size of methods. This shows it's not declarative enough.
The concepts are far too technical: "queue message", "process queue", "watch queue". The concept of "queue" is not clearly introduced in the modelling (i.e. with a model and fields).

Reply:
I don't know what to tell/do here. This method has not change much from previous state: it send a message to the server and handles edge case.
It's not possible the refactor the edge cases here. I guess I could split stuff inside smaller function but that would just move the issue inside a smaller function.

Do you have something to propose ?

Reply:
If it's essentially a move (hard to see with github diff), then let's just keep it like this in the context of this PR.
First sentence was more a remark on method, but it's a move so ignore.
The rest of comment was more about naming of model and the methods, but you already renamed the model from `Sender` to `ThreadMessagePostQueue`, which is much better. You could actually adapt names so they are simpler, because `queue` is redundant with the model name. For instance, `queueMessage` could be `add`... Oh but you actually did it, so it's all great! :-)